elvesAndroid/src/main/java/com/idealista/android/elves/view/Adapter.java

```diff
@@ -10,18 +10,34 @@
 import com.idealista.android.elves.view.widget.OnClicked;
 import java.util.Iterator;
 
-public class Adapter<VItemModel> extends RecyclerView.Adapter<Adapter.Holder> {
+public class Adapter<VItemModel> extends RecyclerView.Adapter<Holder> {
 
+    private static final int TYPE_HEADER = 0;
+    private static final int TYPE_ITEM = 1;
+    private static final int TYPE_FOOTER = 2;
     private final CustomViewCreator<VItemModel> customViewCreator;
     private ListableModel<VItemModel> list;
     private final OnClicked<VItemModel> onClickListener;
+    private CustomView headerCustomView;
+    private CustomView footerCustomView;
+    private boolean hasToShowHeader;
```

Comment:
There is any way to avoid to create a control variable as hasToShowHeader?
Maybe creating a method hasToShowHeader() that use headerCustomView != null or something like this.
Wha do you think?